cli/client.go

```diff
@@ -380,6 +387,111 @@ func RobotsPartShellAction(c *cli.Context) error {
 	)
 }
 
+// checkUpdateResponse holds the values used to hold release information.
+type checkUpdateResponse struct {
+	Name       string `json:"name"`
+	TagName    string `json:"tag_name"`
+	TarballURL string `json:"tarball_url"`
+}
+
+// CheckUpdateAction is the corresponding Action for 'check-update'.
+func CheckUpdateAction(c *cli.Context) error {
+	if c.Bool(quietFlag) {
+		return nil
+	}
+
+	dateCompiledRaw := rconfig.DateCompiled
+	dateCompiled, err := time.Parse("2006-01-02", dateCompiledRaw)
+	if err != nil {
+		// user installed from go install or source
+		utils.UncheckedError(err)
+		return nil
+	}
+
+	conf, err := configFromCache()
+	if err != nil {
+		if !os.IsNotExist(err) {
+			utils.UncheckedError(err)
+			return nil
+		}
+		conf = &config{}
+	}
+
+	if conf.LastUpdateCheck == "" {
+		conf.LastUpdateCheck = time.Now().Format("2006-01-02")
+		err := storeConfigToCache(conf)
+		if err != nil {
+			utils.UncheckedError(err)
+		}
+	} else {
+		lastCheck, err := time.Parse("2006-01-02", conf.LastUpdateCheck)
+		if err != nil {
+			// corrupted date
+			utils.UncheckedError(err)
+			return nil
+		}
+		if time.Since(lastCheck) < time.Hour*24*7 {
+			return nil
+		}
+	}
+
+	// compare to current version
+	appVersion := rconfig.Version
+
+	// skip check if on stable and install is older than two week
+	if appVersion != "" && time.Since(dateCompiled) < time.Hour*24*7*2 {
+		return nil
+	}
+
+	// skip check if on dev/latest and install is less than six weeks old
+	if appVersion == "" && time.Since(dateCompiled) < time.Hour*24*7*6 {
+		return nil
+	}
+
+	localVersion, err := semver.NewVersion(appVersion)
+	if err != nil {
+		warningf(c.App.ErrWriter, "CLI Update Check: failed to parse compiled version: %w", err)
+		return nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, rdkReleaseURL, nil)
+	if err != nil {
+		warningf(c.App.ErrWriter, "CLI Update Check: failed to create request: %w", err)
+		return nil
+	}
+
+	client := http.DefaultClient
+	res, err := client.Do(req)
+	if err != nil {
+		warningf(c.App.ErrWriter, "CLI Update Check: failed to get latest release information: %w", err)
+		return nil
+	}
+
+	resp := checkUpdateResponse{}
+	err = json.NewDecoder(res.Body).Decode(&resp)
+	if err != nil {
+		warningf(c.App.ErrWriter, "CLI Update Check: failed to decode response: %w", err)
+		return nil
+	}
+
+	defer utils.UncheckedError(res.Body.Close())
+
+	latestVersion, err := semver.NewVersion(resp.TagName)
+	if err != nil {
+		warningf(c.App.ErrWriter, "CLI Update Check: failed to parse latest version: %w", err)
+		return nil
+	}
```

Comment:
nit and fairly minor style comment, but I would factor this entire block into a separate function - it's a very self contained/relatively unrelated piece of code that does one specific thing and takes up a good amount of lines, so moving it out of the function means readers don't need to interrupt their flow when reading this larger function